src/components/SaveGameForm.js

```diff
@@ -87,22 +89,70 @@ export default function SaveGameForm({onSubmit}) {
     function handleChange(event) {
         setFormInput({
             ...formInput,
-            [event.target.name]: event.target.value,
+            [event.target.name]: event.target.value
         })
     }
 
     function handleSubmit(event) {
         event.preventDefault()
-        onSubmit(formInput)
+        const validation = validateForm(formInput)
+        if (validation.result) {
+            trimInputs(formInput)
+            onSubmit(formInput)
+            setFormInput({
+                location: '',
+                date: '',
+                players: '',
+                winner: '',
+                shots:'',
+            })
+            setErrorMessage('')
+        } else {
+           setErrorMessage(validation.message)
+        }
+    }
+
+    function validateForm(formInput) {
```

Comment:
I like that you made your own form validation, but I have some extra thoughts to share on that.

I think you should consider validating the user input on a different event, maybe switching input fields or `onChange`, or maybe `onChange` but delayed.
Also, it might be annoying, but the date could use some validation as well, just to ensure your users are not tampering with the data and potentially wrecking stuff. You could do that separately or in another story if you think it is not vital right now, but I would encourage you simply because of how dates are handled here.

Reply:
Regarding the date validation: Are you referring to RegEx in order to check the date? Right now it is really just checking if it is empty or not.